banzai_nres/frames.py

```diff
@@ -99,95 +116,106 @@ def science_fiber(self):
 
     @property
     def traces(self):
-        return self.primary_hdu.traces
+        if 'TRACES' in self._hdu_keys:
+            return self['TRACES'].data
+        else:
+            return self._traces
 
     @traces.setter
     def traces(self, value):
-        self.primary_hdu.traces = value
-
-    @property
-    def num_traces(self):
-        return self.primary_hdu.num_traces
+        self._traces = value
 
     @property
     def background(self):
-        return self.primary_hdu.background
+        return self['BACKGROUND'].data
 
     @background.setter
     def background(self, value):
-        self.primary_hdu.background = value
+        self['BACKGROUND'] = ArrayData(value, name='BACKGROUND')
 
     @property
     def profile(self):
-        return self.primary_hdu.profile
+        if 'PROFILE' in self._hdu_keys:
+            return self['PROFILE'].data
+        else:
+            return self._profile
 
     @profile.setter
     def profile(self, value):
-        self.primary_hdu.profile = value
+        self._profile = value
 
     @property
     def weights(self):
-        return self.primary_hdu.weights
+        if 'WEIGHTS' in self._hdu_keys:
+            return self['WEIGHTS'].data
+        else:
+            return self._weights
 
     @weights.setter
     def weights(self, value):
-        self.primary_hdu.weights = value
+        self._weights = value
 
     @property
     def spectrum(self):
-        return self.primary_hdu.spectrum
+        return self._spectrum
 
     @spectrum.setter
     def spectrum(self, value):
-        self.primary_hdu.spectrum = value
+        self._spectrum = value
+        self['SPECTRUM1D'] = DataTable(value.table, name='SPECTRUM1D')
 
     @property
     def blaze(self):
-        return self.primary_hdu.blaze
+        if 'BLAZE' in self._hdu_keys:
+            return self['BLAZE'].data
+        else:
+            return self._blaze
 
     @blaze.setter
     def blaze(self, value):
-        self.primary_hdu.blaze = value
-
-    @property
-    def features(self):
-        return self.primary_hdu.features
-
-    @features.setter
-    def features(self, value):
-        self.primary_hdu.features = value
-
-    @property
-    def line_list(self):
```

Comment:
How do we handle the line list now? It looks like all the material defining how the line list is handled has been deleted but the parts of wavelength.py that actually fill and use the line list are unchanged. Will this still work with all of the changes in frames.py?

Reply:
Yes. The line list will still work correctly. The only difference is that it is not attached to the output data products anymore.